packages/datadog-instrumentations/src/kafkajs.js

```diff
@@ -52,44 +52,48 @@ addHook({ name: 'kafkajs', file: 'src/index.js', versions: ['>=1.4'] }, (BaseKaf
     const send = producer.send
     const bootstrapServers = this._brokers
 
+    const kafkaClusterIdPromise = getKafkaClusterId(this)
+
     producer.send = function () {
-      const innerAsyncResource = new AsyncResource('bound-anonymous-fn')
-
-      return innerAsyncResource.runInAsyncScope(() => {
-        if (!producerStartCh.hasSubscribers) {
-          return send.apply(this, arguments)
-        }
-
-        try {
-          const { topic, messages = [] } = arguments[0]
-          for (const message of messages) {
-            if (message !== null && typeof message === 'object') {
-              message.headers = message.headers || {}
-            }
+      return kafkaClusterIdPromise.then((clusterId) => {
```

Comment:
If the promise has already resolved, it's unnecessary overhead to call `.then()` on it. Instead you can store the clusterId in its own variable (or just use `this._ddKafkaClusterId`), and if that variable is not set, _then_ you can call `.then()` on the promise.

Reply:
Okay, I changed the code to check if the returned data is a promise, if it is, it calls `.then()`, otherwise continues as normal without calling `.then()`